lib/packwerk/commands.rb

```diff
@@ -0,0 +1,30 @@
+# typed: true
+# frozen_string_literal: true
+
+module Packwerk
+  module Commands
+    extend ActiveSupport::Autoload
+
+    autoload :BaseCommand
+    autoload :CheckCommand
+    autoload :HelpCommand
+    autoload :InitCommand
+    autoload :UpdateCommand, "packwerk/commands/update_todo_command"
+    autoload :UpdateTodoCommand
+    autoload :UsesParseRun
+    autoload :ValidateCommand
+    autoload :VersionCommand
+
+    class << self
+      extend T::Sig
+
+      sig { params(command: String).returns(T.nilable(T.class_of(Commands::BaseCommand))) }
+      def class_for(command)
+        class_name = command.sub(" ", "_").underscore.classify + "Command"
+        if Commands.const_defined?(class_name)
+          Commands.const_get(class_name) # rubocop:disable Sorbet/ConstantsFromStrings
```

Comment:
I think `const_get` is ok, but we don't need this `class_for` method, just lean on autoloading to do this work for you. See my other comment. I ran tests and they all pass with the proposed change.

Reply:
Yeah, so my idea was:
1. I wanted to extract the `const_get` + `rescue` to its own method to make the thing more readable (I didn't want to pollute `Packwerk::Cli#execute_command` with it). It first was a class method on `Packwerk::Cli`, and then naturally moved to `Packwerk::Commands`.
2. Once that was done, I noticed I was using exceptions as control flow whereas there is n appropriate method (`#const_defined?`) to check if the constant exists without raising `NameError`, so I used it instead.

I now realized that when I moved the method, I could have omitted the `Commands` here:

```diff
       sig { params(command: String).returns(T.nilable(T.class_of(Commands::BaseCommand))) }
       def class_for(command)
         class_name = command.sub(" ", "_").underscore.classify + "Command"
-        if Commands.const_defined?(class_name)
-          Commands.const_get(class_name) # rubocop:disable Sorbet/ConstantsFromStrings
+        if const_defined?(class_name)
+          const_get(class_name) # rubocop:disable Sorbet/ConstantsFromStrings
```

Reply:
> I didn't want to pollute `Packwerk::Cli#execute_command` with it

Yeah but now we have a new public method to support, just for a single use case. Feels to me more appropriate just to hide the implementation in the only place that actually uses it (esp since it's just a couple lines).

> I noticed I was using exceptions as control flow

Calling an undefined command _is_ an "exception" though, so feels to me that rescuing here is fine (to avoid exposing an exception to the user). I would consider it "control flow" if we were then doing something other work in the `rescue`, but we're literally just cleaning up the output.

Reply:
>  a new public method to support
That's a very fair point! Maybe I can move the method back to `Cli`, make it private, and switch back to rescuing `NameError`. I don't have a strong opinion on this bit.